File: algorithms/gnome_sort/java/GnomeSort.java

public class GnomeSort {
	
	
	public GnomeSort(int[] list) {
		
		for (int index = 1; index < list.length;) {
			
			if (list[index - 1] <= list[index]) {
				index++;
			} else {
				int tempVal = list[index];
				list[index] = list[index - 1];
				list[index - 1] = tempVal;
				index--;
				if (index == 0) {
					index = 1;
				}
			}
		}
	}
}

File: algorithms/gnome_sort/gnome_sort.py
__author__ = "Matteo Veraldi"
__email__ = "mattveraldi@gmail.com"
__github__ = "mattveraldi"

def gnome_sort(lis):
    pos = 0
    while(pos < len(lis)):
        if(pos == 0 or lis[pos] >= lis[pos-1]):
            pos += 1
        else:
            lis[pos], lis[pos-1] = lis[pos-1], lis[pos]
            pos -= 1
    return lis

def test():
    test_list = [119, 18, 51, 11, 9, 8.5]
    expected = [8.5, 9, 11, 18, 51, 119]
    print("###GNOME SORT by Matteo Veraldi (github.com/mattveraldi)###")
    assert gnome_sort(test_list) == expected, "test failed."
    print("test ok.")
